BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -249,37 +293,74 @@ protected override void AutoPilot(FlightCtrlState s)
 
         void UpdateStatus()
         {
+            // Update propulsion and weapon status
             bool hasRCSFore = VesselModuleRegistry.GetModules<ModuleRCS>(vessel).Any(e => e.rcsEnabled && !e.flameout && e.useThrottle);
             hasPropulsion = hasRCSFore || VesselModuleRegistry.GetModuleEngines(vessel).Any(e => (e.EngineIgnited && e.isOperational));
             hasWeapons = weaponManager.HasWeaponsAndAmmo();
 
+            // Update intervals
             if (weaponManager.incomingMissileVessel != null && updateInterval != emergencyUpdateInterval)
                 updateInterval = emergencyUpdateInterval;
             else if (weaponManager.incomingMissileVessel == null && updateInterval == emergencyUpdateInterval)
                 updateInterval = combatUpdateInterval;
 
-            if (vessel.isActiveVessel && targetVessel && (vessel.targetObject == null || vessel.targetObject.GetVessel() != targetVessel))
+            // Set evasion status
+            EvasionStatus();
+
+            // Set target as UI target
+            if (vessel.isActiveVessel && targetVessel && !targetVessel.IsMissile() && (vessel.targetObject == null || vessel.targetObject.GetVessel() != targetVessel))
             {
-                FlightGlobals.fetch.SetVesselTarget(targetVessel);
+                FlightGlobals.fetch.SetVesselTarget(targetVessel, true);
             }
 
             return;
         }
 
+        void EvasionStatus()
+        {
+            // Check if we should be evading gunfire, missile evasion is handled separately
+            threatRating = evasionThreshold + 1f; // Don't evade by default
+            evadingGunfire = false;
+            if (weaponManager != null && weaponManager.underFire)
+            {
+                if (weaponManager.incomingMissTime >= evasionTimeThreshold && weaponManager.incomingThreatDistanceSqr >= evasionMinRangeThreshold * evasionMinRangeThreshold) // If we haven't been under fire long enough or they're too close, ignore gunfire
+                    threatRating = weaponManager.incomingMissDistance;
+            }
+            // If we're currently evading or a threat is significant
+            if ((evasiveTimer < minEvasionTime && evasiveTimer != 0) || threatRating < evasionThreshold)
+            {
+                if (evasiveTimer < minEvasionTime)
+                {
+                    threatRelativePosition = vessel.Velocity().normalized + vesselTransform.right;
+                    if (weaponManager)
+                    {
+                        if (weaponManager.underFire)
+                            threatRelativePosition = weaponManager.incomingThreatPosition - vesselTransform.position;
+                    }
+                }
+                evadingGunfire = true;
```

Comment:
Should `evadingGunfire` be set `true` if it's already evading a missile? If not, then that should probably be checked for.